public/app/plugins/panel/flamegraph/flamegraphV2/components/FlameGraph/rendering.ts

```diff
@@ -184,28 +204,36 @@ export function renderRect(
   ctx.beginPath();
   ctx.rect(rect.x + (rect.collapsed ? 0 : BAR_BORDER_WIDTH), rect.y, rect.width, rect.height);
 
-  //  / (rangeMax - rangeMin) here so when you click a bar it will adjust the top (clicked)bar to the most 'intense' color
-  const intensity = Math.min(1, rect.ticks / totalTicks / (rangeMax - rangeMin));
-  const h = 50 - 50 * intensity;
-  const l = 65 + 7 * intensity;
-
-  const name = rect.label;
-
-  if (!rect.collapsed) {
-    ctx.stroke();
+  const color =
+    colorScheme === ColorScheme.ValueBased
+      ? getBarColorByValue(rect.ticks, totalTicks, rangeMin, rangeMax)
+      : getBarColorByPackage(rect.label, theme);
 
-    if (foundNames) {
-      ctx.fillStyle = foundNames.has(name) ? getBarColor(h, l) : colors[55];
+  if (foundNames) {
```

Comment:
Just refactoring to separate decision on colors from how to render and simplify the Ifs a bit that way.